pkg/reconciler/intevents/topic/topic.go

```diff
@@ -91,6 +91,12 @@ var _ topicreconciler.Interface = (*Reconciler)(nil)
 func (r *Reconciler) ReconcileKind(ctx context.Context, topic *v1.Topic) reconciler.Event {
 	ctx = logging.WithLogger(ctx, r.Logger.With(zap.Any("topic", topic)))
 
+	if v, present := topic.Annotations[v1.LoggingE2ETestAnnotation]; present {
+		// This is added purely for the TestCloudLogging E2E tests, which verify that the log line
+		// is written if this annotation is present.
+		logging.FromContext(ctx).Desugar().Error("Adding log line for the TestCloudLogging E2E tests", zap.String(v1.LoggingE2EFieldName, v))
```

Comment:
We could probably add a way to turn it off. E.g. if a specific ConfigMap is not present or a key doesn't have a specific value, then don't log anything. For this in particular, I don't think it is needed because the cost of getting it wrong is so low. 

However, if we are going to be doing experimental things, then we should probably have a framework for doing them. And these log lines could follow that framework.

Reply:
How about having a helper function for all the extra code? IIUC, all the added code can be done in one or two functions, eg map vs env. Then we can use e2e build tag in the helper functions to swap implementation.

Reply:
I am leaning the helper function way as well. It'll save us some refactoring in the future.

Reply:
> Then we can use e2e build tag in the helper functions to swap implementation.

I don't want this to be a build time flag, because I want this test to run in production. Failure of this test will prevent a release from going out. Its whole purpose is to remove a manual test that we require today.

I did create helper functions to centralize almost all the annotation/env var manipulation.